MapleServer2/PacketHandlers/Game/GuildHandler.cs

```diff
@@ -418,14 +436,36 @@ private void HandleGuildListRequest(GameSession session, PacketReader packet)
 
         private void HandleUseBuff(GameSession session, PacketReader packet)
         {
-            int buffID = packet.ReadInt();
+            int buffId = packet.ReadInt();
 
-            if (!session.Player.Wallet.Meso.Modify(-100000))
+            Guild guild = GameServer.GuildManager.GetGuildById(session.Player.GuildId);
+            if (guild == null)
             {
                 return;
             }
 
-            session.Send(GuildPacket.ActivateBuff(buffID));
+            GuildBuff buff = GuildMetadataStorage.GetBuff(buffId);
+
+            if (buff.Id > 1000)
+            {
+                if (!session.Player.Wallet.Meso.Modify(-buff.Cost))
+                {
+                    return;
+                };
+            }
+            else
+            {
+                if (buff.Cost > guild.Funds)
+                {
+                    return;
+                }
+                else
+                {
+                    guild.Funds -= buff.Cost;
+                }
+            }
```

Comment:
The bottom part can look something like this:
```
}
else if (buff.Cost > guild.Funds)
{
    return;
}
               
guild.Funds -= buff.Cost;
```

Reply:
Wouldn't this mean if someone tries to use a buff of an id lower than 1000, they're also paying the cost with guild funds?

Reply:
Yeah if buff id < 1000 they will also pay with the guild funds, aside from that you could remove the else just to make the code cleaner.
Something like this:
```
if (buff.Cost > guild.Funds)
{
    return;
}
 guild.Funds -= buff.Cost;
                
```